### ISSUE TITLE
Markdown anchors do not work with special symbols in path

### ISSUE THREAD

**Issue by eseiler** [doxygen/doxygen#8156]
**Describe the bug**
Have at least two markdown files that just have a header with an anchor
```markdown
# Header 1 {#anchor1}
```
```markdown
# Header 2 {#anchor2}
```

Additionally, the Doxyfile is in another directory that has the directory of the markdown files as INPUT.
A simple version looks like
```console
.
├── build
│   └── Doxyfile
├── md1.md
└── md2.md
```

If the path to the markdown files contains `.` or `+`, a warning will be emitted, e.g.
```console
/dev/shm/example_1.0/md2.md:1: warning: multiple use of section label 'md__dev_shm_example_1' while adding anchor, (first occurrence: /dev/shm/example_1.0/md1.md, line 1)
```

**Expected behavior**
No warning, anchor works as expected.

**To Reproduce**
[example_1.0.tar.gz](https://github.com/doxygen/doxygen/files/5483297/example_1.0.tar.gz)
[example+ds.tar.gz](https://github.com/doxygen/doxygen/files/5483299/example%2Bds.tar.gz)
[example_1.0+ds.tar.gz](https://github.com/doxygen/doxygen/files/5483298/example_1.0%2Bds.tar.gz)

All cases trigger the bug, the INPUT path assumes `/dev/shm/example_1.0` etc.
If you extract to a different path, adjust the `INPUT` in the `build/Doxyfile`.

**Version**
Happens in 1.8.20 and on master.
Seems to have worked in 1.8.17 (did not confirm).

**Additional context**
https://github.com/seqan/seqan3/issues/2011

**Comment by albert-github** [comment]
Problem surfaced first time in version 1.8.19.

I've just pushed a proposed patch, pull request #8157

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
